src/views/DEX/components/HistoryContainer/Components/OverallOrderHistory/OverallOrderHistory.tsx

```diff
@@ -18,27 +18,18 @@ const OverallOrderHistory: React.FC<IOverallOrderHistoryProps> = ({ assetId }) =
   const { getTrades } = useAPI();
   const { blockHeight } = useBlocks();
 
-  // set the trade history for the current asset
-  useEffect(() => {
-    async function fetchTrades() {
-      if (getTrades === undefined || assetId === undefined) {
-        return;
-      }
-
-      try {
-        const result = await getTrades(assetId);
-
-        if (result) {
-          setTradeHistory(result);
-        }
-      } catch (e) {
-        console.error("error while getting trade history in DEX component:", e);
-        return;
-      }
+  const fetchTrades = useCallback(async () => {
+    if (getTrades === undefined || assetId === undefined) {
+      return;
     }
 
+    setTradeHistory(await getTrades(assetId));
```

Comment:
`tradeHistory` can be undefined and that's what's returned here when `getTrades` has a failure. this is an instance where the error handling happens in a higher level and only is indicated by 'undefined' being returned here, which is admittedly ambiguous but the user is still warned of the error, which may inform them enough.

Reply:
ok, if undefined is the only thing that can be returned that's fine. undefined could either be returned if the user has no trade history or if things error, and I think we would only report the snackbar on the existence of a `.error` prop so that's acceptable behavior (I didn't re-read to confirm this but I'm pretty sure that's how it works based on some testing I did earlier).